packages/components/src/Filter/FilterBar/_docs/FilterBar.spec.stories.tsx

```diff
@@ -0,0 +1,208 @@
+  argTypes: {
+    filters: { control: false },
+    values: { control: false },
+    onValuesChange: { control: false },
+  },
+  args: {
+    filters: [], // Defined in stories
+    values: {}, // Defined in stories
+    onValuesChange: fn(),
+  },
+} satisfies Meta<typeof FilterBar>
+
+export default meta
+
+type Story = StoryObj<typeof meta>
+
+type Values = {
+  flavour: string
+  deliveryDates: DateRange
+  toppings: string[]
+  drank: Date
+}
+
+const filters = [
+  {
+    id: "flavour",
+    name: "Flavour",
+    Component: (
+      <FilterBar.Select
+        items={[
+          { value: "jasmine-milk-tea", label: "Jasmine Milk Tea" },
+          { value: "honey-milk-tea", label: "Honey Milk Tea" },
+          { value: "lychee-green-tea", label: "Lychee Green Tea" },
+        ]}
+      />
+    ),
+  },
+  {
+    id: "deliveryDates",
+    name: "Delivery Dates",
+    Component: <FilterBar.DateRangePicker />,
+  },
+  {
+    id: "toppings",
+    name: "Toppings",
+    Component: (
+      <FilterBar.MultiSelect
+        items={[
+          { value: "none", label: "None" },
+          { value: "pearls", label: "Pearls" },
+          { value: "fruit-jelly", label: "Fruit Jelly" },
+          { value: "peanuts", label: "Peanuts" },
+          { value: "coconut", label: "Coconut" },
+          { value: "aloe", label: "Aloe Vera" },
+          { value: "mochi", label: "Mini Mochi" },
+          { value: "popping-pearls", label: "Popping Pearls" },
+        ]}
+      >
+        {(): JSX.Element => (
+          <>
+            <FilterMultiSelect.SearchInput />
+            <FilterMultiSelect.ListBox>
+              {({ allItems }): JSX.Element | JSX.Element[] =>
+                allItems.map(item => (
+                  <FilterMultiSelect.Option key={item.key} item={item} />
+                ))
+              }
+            </FilterMultiSelect.ListBox>
+            <FilterMultiSelect.MenuFooter>
+              <FilterMultiSelect.SelectAllButton />
+              <FilterMultiSelect.ClearButton />
+            </FilterMultiSelect.MenuFooter>
+          </>
+        )}
+      </FilterBar.MultiSelect>
+    ),
+    isRemovable: true,
+  },
+  {
+    id: "drank",
+    name: "Drank",
+    Component: <FilterBar.DatePicker />,
+    isRemovable: true,
+  },
+] satisfies Filters<Values>
+
+export const ClearAll: Story = {
+  render: args => {
+    const [activeValues, onActiveValuesChange] = useState<Partial<Values>>({})
+
+    return (
+      <FilterBar<Values>
+        {...args}
+        filters={filters}
+        values={activeValues}
+        onValuesChange={onActiveValuesChange}
+      />
+    )
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement.parentElement!)
+
+    await step(
+      "Clear all button hidden by default given no values",
+      async () => {
+        expect(
+          canvas.queryByRole("button", {
+            name: "Clear all filters",
+          })
+        ).toBe(null)
```

Comment:
queryBy's whole thing is to return null when the element isn't found. Asserting that null is not in the document does pass, but doesn't make sense.

https://testing-library.com/docs/guide-disappearance/#asserting-elements-are-not-present

Reply:
> The [jest-dom](https://testing-library.com/docs/ecosystem-jest-dom) utility library provides the .toBeInTheDocument() matcher, which can be used to assert that an element is in the body of the document, or not. This can be more meaningful than asserting a query result is null.

https://testing-library.com/docs/guide-disappearance/#nottobeinthedocument

Right under it? :)

Reply:
Just read further down and they say that asserting `not.toBeInTheDocument()` can be more meaningful. Can't say I agree but much of a muchness I'll switch to it